src/xref2/compile.ml

```diff
@@ -673,33 +597,33 @@ and type_expression_object env o =
   { o with fields = List.map field o.fields }
 
 and type_expression_package env p =
-  let exception
-    Compile_type_expression_package of Tools.signature_of_module_error option
-  in
   let open TypeExpr.Package in
   let cp = Component.Of_Lang.(module_type_path empty p.path) in
   match Tools.lookup_and_resolve_module_type_from_path true env cp with
-  | Resolved (path, mt) ->
-      let sg =
-        match Tools.signature_of_module_type env mt with
-        | Ok sg -> sg
-        | Error e -> raise (Compile_type_expression_package (Some e))
-      in
-      let substitution (frag, t) =
-        let cfrag = Component.Of_Lang.(type_fragment empty frag) in
-        let frag' =
-          match
-            Tools.resolve_mt_type_fragment env (`ModuleType path, sg) cfrag
-          with
-          | Some cfrag' -> Lang_of.(Path.resolved_type_fragment empty) cfrag'
-          | None -> raise (Compile_type_expression_package None)
-        in
-        (`Resolved frag', type_expression env t)
-      in
-      {
-        path = module_type_path env p.path;
-        substitutions = List.map substitution p.substitutions;
-      }
+  | Resolved (path, mt) -> (
+      match Tools.signature_of_module_type env mt with
+      | Error _ ->
+          lookup_failure ~what:(`Package cp) `Lookup;
```

Comment:
Details about the error are lost. Should I try to keep them ? It happens in a few other places.

Reply:
Probably fine for now